disnake/onboarding.py

```diff
@@ -0,0 +1,239 @@
+# SPDX-License-Identifier: MIT
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Sequence, Union
+
+from .emoji import Emoji, PartialEmoji, _EmojiTag
+from .enums import OnboardingPromptType, try_enum
+from .mixins import Hashable
+from .object import Object
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .abc import Snowflake
+    from .guild import Guild, GuildChannel
+    from .role import Role
+    from .types.onboarding import (
+        Onboarding as OnboardingPayload,
+        OnboardingPrompt as OnboardingPromptPayload,
+        OnboardingPromptOption as OnboardingPromptOptionPayload,
+    )
+
+__all__ = (
+    "Onboarding",
+    "OnboardingPrompt",
+    "OnboardingPromptOption",
+)
+
+
+class Onboarding:
+    """Represents a guild onboarding.
+
+    .. versionadded:: 2.9
+
+    Attributes
+    ----------
+    guild: :class:`Guild`
+        The guild this onboarding is part of.
+    prompts: List[:class:`OnboardingPrompt`]
+        The onboarding prompts.
+    enabled: :class:`bool`
+        Whether onboarding is enabled.
+    """
+
+    __slots__ = (
+        "guild",
+        "prompts",
+        "enabled",
+        "_guild_id",
+        "_default_channel_ids",
+    )
+
+    def __init__(self, *, guild: Guild, data: OnboardingPayload):
+        self.guild = guild
+        self._from_data(data)
+
+    def _from_data(self, data: OnboardingPayload):
+        self.prompts = [
+            OnboardingPrompt._from_dict(data=prompt, guild=self.guild) for prompt in data["prompts"]
+        ]
+        self.enabled = data["enabled"]
+        self._default_channel_ids = list(map(int, data["default_channel_ids"]))
+
+    def __repr__(self) -> str:
+        return (
+            f"<Onboarding guild_id={self.guild.id!r} prompts={self.prompts!r}"
+            f" default_channel_ids={self._default_channel_ids!r} enabled={self.enabled!r}>"
+        )
```

Comment:
Instead of the guild ID, this should include the guild itself.
Additionally, `default_channel_ids` is a bit too verbose to include here.

Reply:
dae932be51d4522f81a40a11a416f251f3bc2354